services/algolia-sync/src/components/algolia/site-sync.js

```diff
@@ -0,0 +1,81 @@
+const gql = require('graphql-tag');
+const algoliasearch = require('algoliasearch');
+const cliProgress = require('cli-progress');
+const apollo = require('../graphql/query');
+const helpers = require('./helpers');
+const { ALGOLIA_APPID, ALGOLIA_API_KEY } = require('../../env');
+
+const client = algoliasearch(ALGOLIA_APPID, ALGOLIA_API_KEY);
+
+const bar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+
+const query = gql`
+  query websiteScheduledContent($input: AllPublishedContentQueryInput!) {
+    allPublishedContent(input: $input){
+      pageInfo {
+        hasNextPage
+      }
+      edges {
+        node {
+          id
+          type
+          created
+          published
+          updated
+          status
+          websiteSchedules { section { hierarchy { fullName } }}
+          primaryImage { name filePath fileName src isLogo }
+          primarySite{ host }
+          name
+          teaser
+        }
+      }
+    }
+  }
+`;
+
+const total = () => (gql`
+  query websiteScheduledContent($input: AllPublishedContentQueryInput!) {
+    allPublishedContent(input: $input){
+      totalCount
+    }
+  }
+`);
+
+const upsertToIndex = async (tenant, limit, skip, totalContent) => {
+  bar.update(skip);
+
+  const input = {
+    pagination: {
+      limit,
+      skip,
+    },
+  };
+
+  const data = await apollo.queryFromBase(query, tenant, input);
+  const nodes = data.allPublishedContent.edges.map(content => content.node);
+  const algoliaData = await helpers.buildObj(nodes, tenant);
+  client.multipleBatch(algoliaData, (err) => { if (err) throw err; });
+
+  if (skip < totalContent) {
+    upsertToIndex(tenant, limit, (skip + limit), totalContent);
```

Comment:
Are you purposely not awaiting this function?? If you don't await the function will end immediately regardless of the upsert result - and it will never throw an error if it encounters one. Either change to 

```js
await upsertToIndex(tenant, limit, (skip + limit), totalContent);
```
Or 
```js
upsertToIndex(tenant, limit, (skip + limit), totalContent).catch(newrelic.noticeError.bind(newrelic));
```

I'm pretty sure you _do_ want to await this though...

Reply:
Ya it should probably wait before starting the next batch. I went ahead and updated it as you suggested.